### ISSUE TITLE
Add option in Source Search to search in namespace files

### ISSUE THREAD

**Issue by renemolenaar82** [kestra-io/kestra#5728]
### Feature description

I use [Python Command](https://kestra.io/plugins/plugin-script-python/tasks/io.kestra.plugin.scripts.python.commands) more often than [Python Script ](https://kestra.io/plugins/plugin-script-python/tasks/io.kestra.plugin.scripts.python.script) because it keeps the main flow nice and clean. Python code is stored in multiple `.py` namespace files.

Source Search is great to quickly find all flows where you used a specific task, variable, etc. 

It would be useful if you could let it search in namespace files as well. This way, it would be possible to search the entire codebase, not just the flows.

**Comment by anna-geller** [comment]
for the engineer who would pick this up: let's add a "Namespace Files" filter (toggle or a dropdown) with the option to optionally also search across Namespace Files

![image](https://github.com/user-attachments/assets/c3ce39d9-45bf-4d7d-b61b-7ab1a65b72f2)

**Comment by Malaydewangan09** [comment]
@anna-geller 
What kind of response can be expected when "Namespace Files" will be turned on or selected?
Will it include code segments from the matching file, similar to how flows includes flow segments?

**Comment by anna-geller** [comment]
Yes, that sounds very reasonable

**Comment by Malaydewangan09** [comment]
@anna-geller i would love to pick this up
But, given that this involves searching the context in all files, could you please provide a brief overview of how we intend to implement it?

**Comment by anna-geller** [comment]
It's up to you how to implement it, you can check how namespace files are handled so far in the Editor.

I do agree though that this is a fairly complex feature as you would need to search across multiple files which are not stored in kestra directly but rather in internal storage (like e.g. S3). For security reasons, kestra doesn't store your code directly, it fetches it from internal storage when needed

**Comment by Malaydewangan09** [comment]
Yes, I looked into the internal storage and how namespace files are handled.
I will get back to you with an implementation idea in some time.
Thanks!

**Comment by loicmathieu** [comment]
Superseded by https://github.com/kestra-io/kestra-ee/issues/8644
